test/Microsoft.TestPlatform.AcceptanceTests/DiscoveryTests.cs

```diff
@@ -91,5 +97,33 @@ public void DiscoverTestsShouldShowProperWarningIfNoTestsOnTestCaseFilter(Runner
 
             this.ExitCodeEquals(0);
         }
+
+        [TestMethod]
+        public void TypesToLoadAttributeTests()
+        {
+            var environment = new IntegrationTestEnvironment();
+            var extensionsDirectory = environment.ExtensionsDirectory;
+            var extensionsToVerify = new Dictionary<string, string[]>
+            {
+                {"Microsoft.TestPlatform.Extensions.EventLogCollector.dll", new[] { "Microsoft.TestPlatform.Extensions.EventLogCollector.EventLogDataCollector"} },
+                {"Microsoft.TestPlatform.Extensions.BlameDataCollector.dll", new[] { "Microsoft.TestPlatform.Extensions.BlameDataCollector.BlameLogger", "Microsoft.TestPlatform.Extensions.BlameDataCollector.BlameCollector" } },
+                {"Microsoft.VisualStudio.TestPlatform.Extensions.Html.TestLogger.dll", new[] { "Microsoft.VisualStudio.TestPlatform.Extensions.HtmlLogger.HtmlLogger" } },
+                {"Microsoft.VisualStudio.TestPlatform.Extensions.Trx.TestLogger.dll", new[] { "Microsoft.VisualStudio.TestPlatform.Extensions.TrxLogger.TrxLogger" } },
+                {"Microsoft.TestPlatform.TestHostRuntimeProvider.dll", new[] { "Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DefaultTestHostManager", "Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Hosting.DotnetTestHostManager" } }
+            };
+
+            foreach (var extension in extensionsToVerify.Keys)
+            {
+                var assemblyFile = Path.Combine(extensionsDirectory, extension);
+                var assembly = Assembly.LoadFrom(assemblyFile);
+
+                var expected = extensionsToVerify[extension];
+                var actual = TypesToLoadUtilities.GetTypesToLoad(assembly).Select(i => i.FullName).ToArray();
+
+                CollectionAssert.AreEquivalent(expected, actual, $"Specified types using TypesToLoadAttribute in \"{extension}\" assembly doesn't match the expected.");
+            }
+
+            Assert.IsTrue(true);
```

Comment:
Why we need this?

Reply:
This test verifies the logic works as expected.